federated-identity/src/main/java/com/iluwatar/federatedIdentity/Consumer.java

```diff
@@ -0,0 +1,175 @@
+package com.iluwatar.federatedIdentity;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import org.apache.http.client.fluent.Request;
+
+/**
+ * act as consumer.
+ *
+ * <p>@author jasciiz
+ */
+public class Consumer {
+  String name;
+  int age;
+  Date data;
+  String others;
+  static String IdP = "http://localhost:7001/IdP";
+  static String Server = "http://localhost:7003/Server";
+  static SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+  boolean isRegister;
+  boolean isLand;
+  String claim;
+
+  /**
+   * construct with parameters.
+   *
+   * <p>@param name the name of consumer
+   * @param age the age of consumer
+   * @param data the date of birth of consumer
+   * @param others the other information of consumer
+   * @throws IOException throw
+   *  when input name with "//".
+   */
+  public Consumer(String name, int age, String data, String others) throws IOException {
+    if(name.contains("//")){
+      throw new IOException("wrong name! Do not contain \"//\"!");
+    }
+    this.name = name;
+    try{
+      this.data = sdf.parse(data);
+    }catch (ParseException e){
+      e.printStackTrace();
+    }
+    this.others = others;
+    this.age = age;
+    this.isRegister = false;
+    this.claim = null;
+    this.isLand = false;
+  }
+
+  /**
+   * construct without parameter.
+   * <p>
+   */
+  public Consumer(){
+    this.name = "iluwater";
+    this.age = 2020-1990;
+    this.data = new Date(1990-1900, Calendar.DECEMBER,8);
+    this.others = "hello world";
+    this.isRegister = false;
+    this.claim = null;
+    this.isLand = false;
+  }
+
+  /**
+   * get IdP link.
+   * <p>
+   * @return
+   */
+  public String getIdP() {
+    return IdP;
+  }
+
+  /**
+   * set new IdP link.
+   * <p>
+   * @param IdP the new link
+   */
+  public void setIdP(String IdP) {
+    Consumer.IdP = IdP;
+  }
+
+  /**
+   * resister self in IdP.
+   * <P>
+   * @exception IOException throw
+   * when no get response from IdP.
+   */
+  public void register() {
+    StringBuilder stringBuilder = new StringBuilder();
```

Comment:
In Java 11 we can use `var` to declare variables like this. See https://docs.oracle.com/en/java/javase/13/language/local-variable-type-inference.html